packages/langium/src/grammar/generated/ast.ts

```diff
@@ -418,7 +411,7 @@ export function isGroup(item: unknown): item is Group {
 }
 
 export interface Keyword extends AbstractElement {
-    readonly $container: Alternatives | Assignment | CharacterRange | CrossReference | Grammar | Group | NegatedToken | ParserRule | TerminalAlternatives | TerminalGroup | TerminalRule | UnorderedGroup | UntilToken;
+    readonly $container: CharacterRange;
```

Comment:
This looks wrong: CharacterRange is not the only possible container of Keyword.

Reply:
@spoenemann, it's correctly for the current calculation:
* `Keyword` is used in the assignment in the only one place -- in `CharacterRange`
* One more usage is the rule `PredicatedKeyword` that infers `Keyword` -- 1) the interface `PredicatedKeyword` will not be generated by the grammar definition, so no its income to container types; 2) if we would have the interface `PredicatedKeyword`, it would be a child of `Keyword` -- no income to container type, too, because we don't lift children container types in parents anymore
* Other usages are in type alternatives and are ignored by the new algorithm: `AssignableTerminal`, `CrossRefTerminal`, `AbstractTerminal`

Reply:
There's a rather complex chain of unassigned rule calls here:

Alternatives > ConditionalBranch > UnorderedGroup > Group > AbstractToken > AbstractTokenWithCardinality > AbstractTerminal > Keyword

Alternatives has two possible containers: Group and ParserRule. Keyword should inherit those containers.

I've seen that in simpler examples of unassigned rule calls, the container types are fine. So maybe this is just a special case for which we can create an issue, but we ignore it for now?

Reply:
By my understanding it's not an issue, but an expected behaviour for the current algorithm.

The moment is the algorithm is working with inferred / declared types (from the `ast.ts`, basically) and not with parser rules content. That means, not all information that you can see in the `.langium` is available for the algorithm. As a result:
1. Most of types with the names from the chain doesn't exist
2. `Group` and `ParserRule` contain only `AbstractElement` and not `Alternatives`

Reply:
Ok let's ignore the issue for now and get back to it in case it turns out to be a real problem. I created #1063 so we don't forget.